tests/integration/security/cacert_rotation/main_test.go

```diff
@@ -126,6 +126,59 @@ func TestReachability(t *testing.T) {
 
 				// retry when workload cert is not updated
 				if updateTime.After(lastUpdateTime) {
+					lastUpdateTime = updateTime
+					t.Logf("workload cert is updated, last update time: %v", updateTime)
+					return true
+				}
+
+				return false
+			}, retry.Timeout(5*time.Minute), // Need more time for waiting, see https://github.com/istio/istio/pull/47839#issuecomment-1809510888
+				retry.Delay(1*time.Second))
+
+			// step 2: Update CA signing key/cert with cacert to trigger workload cert resigning
+			if err := cert.CreateCustomCASecret(t,
+				"ca-cert-alt.pem", "ca-key-alt.pem",
+				"cert-chain-alt.pem", "root-cert-combined-2.pem"); err != nil {
+				t.Errorf("failed to update CA secret: %v", err)
+			}
+
+			// Wait for workload cert to be updated
+			retry.UntilOrFail(t, func() bool {
+				updateTime, err := getWorkloadCertLastUpdateTime(t, from[0], istioCtl)
+				if err != nil {
+					t.Logf("failed to get workload cert last update time: %v", err)
+					return false
+				}
+
+				// retry when workload cert is not updated
+				if updateTime.After(lastUpdateTime) {
+					lastUpdateTime = updateTime
+					t.Logf("workload cert is updated, last update time: %v", updateTime)
+					return true
+				}
+
+				return false
+			}, retry.Timeout(5*time.Minute), // Need more time for waiting, see https://github.com/istio/istio/pull/47839#issuecomment-1809510888
+				retry.Delay(1*time.Second))
```

Comment:
looks these code is repeated

Reply:
They are necessary to check after each step, the certs should be resigned. I guess we can abstract a function